atc/db/migration/migrations/1557152441_separate_worker_task_cache.up.sql

```diff
@@ -0,0 +1,52 @@
+BEGIN;
+  CREATE TABLE task_caches (
+    id serial,
+    worker_name text,
+    job_id integer,
+    step_name text NOT NULL,
+    path text NOT NULL,
+    PRIMARY KEY ("id")
+  );
+  CREATE UNIQUE INDEX task_caches_job_id_step_name_path_uniq
+    ON task_caches (job_id, step_name, path);
+
+  CREATE INDEX task_caches_job_id ON task_caches USING btree (job_id);
+
+  ALTER TABLE ONLY task_caches
+    ADD CONSTRAINT task_caches_job_id_fkey FOREIGN KEY (job_id) REFERENCES jobs(id) ON DELETE CASCADE;
+
+  ALTER TABLE worker_task_caches
+    ADD COLUMN task_cache_id INTEGER;
+
+  ALTER TABLE worker_task_caches
+    ADD CONSTRAINT worker_task_caches_task_cache_fkey FOREIGN KEY (task_cache_id) REFERENCES task_caches (id) ON DELETE CASCADE;
+  CREATE INDEX worker_task_caches_task_cache_id ON worker_task_caches (task_cache_id);
+
+  WITH ins AS (
```

Comment:
Not sure if you want to but maybe a migration test might be useful for this query that migrates data (along with the down migration one)

Reply:
I have been thinking about this, IMHO the test here might not be very much needed:

1. The worker_task_caches table will be removed by runtime eventually 
2. The table is relatively simple 
3. The task cache data is not a key requirement for concourse being functioning properly. If in worst case all task cache data are lost during up/down migration, the task can still run (just that it will need to populate the task cache path again) 

What do you think @clarafu ? Thx!

Reply:
Makes sense to me, it's just my personal preference of usually writing tests for migrations that deal with moving data over but in this case I can see it being unnecessary. 👍